src/components/ProductListPage/ProductListPage.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState, useRef } from "react";
+import productCategories from "../../mocks/en-us/product-categories.json";
+import productsMock from "../../mocks/en-us/products.json";
+import Card from "../Card/Card";
+
+const ProductListPage = () => {
+  const [isLoading, setLoading] = useState(true);
+  const initProducts = productsMock.results;
+  const [products, setProducts] = useState(initProducts);
+  const [filters, setFilters] = useState(new Map());
```

Comment:
For this specific case, we can only use simple arrays, because you can only save the selected categories in the array, When the users disable one you only need to remove the value, right now `Map` is over-engineering, in the practice, the devs want to keep simple KISS https://en.wikipedia.org/wiki/KISS_principle. Map is for cases when we want to store the key-value and the values an be different for some items.

<img width="713" alt="Screen Shot 2023-09-07 at 15 29 06" src="https://github.com/dbracamonteswz/ondemand-react-bootcamp/assets/96157153/83d154f3-0a08-4e29-ba8f-c91e6e840df9">

Reply:
done